Add module for managing boot environments on FreeBSD/Solaris/illumos …

xen0l opened the pull request:
##### ISSUE TYPE

<!--- Pick one below and delete the rest: -->
- New Module Pull Request
##### COMPONENT NAME

beadm.py
##### ANSIBLE VERSION

<!--- Paste verbatim output from “ansible --version” between quotes below -->

```
ansible 2.1.2.0
  config file = 
  configured module search path = Default w/o overrides
```
##### SUMMARY

This module enables management of ZFS boot environments on FreeBSD/illumos systems (Solaris should be supported as well, but it wasn't tested).

xen0l committed:
Add module for managing boot environments on FreeBSD/Solaris/illumos systems

xen0l commented:
Migrated from https://github.com/ansible/ansible-modules-extras/pull/3076

xen0l commented:
@ansible, I got this reviewed. Can this be merged?

xen0l commented:
@sivel you closed ansible/ansible-modules-extras#3076, which was reviewed there. This was reviewed by @jurajlutter there. Any chance to get this merged?

gundalow commented on lib/ansible/modules/system/beadm.py:
```diff
@@ -0,0 +1,376 @@
+description:
+    - Create, delete or activate ZFS boot environments.
+    - Mount and unmount ZFS boot environments.
+version_added: "2.2"
```
2.3

gundalow commented on lib/ansible/modules/system/beadm.py:
```diff
@@ -0,0 +1,376 @@
+'''
+
+EXAMPLES = '''
+# Create ZFS boot environment
```
Please write examples in multi-line YAML with `key: value`
Also  use `name: ...` rather than a comment to describe what the task is doing.

gundalow commented on lib/ansible/modules/system/beadm.py:
```diff
@@ -0,0 +1,376 @@
+    module.exit_json(**result)
+
+
+from ansible.module_utils.basic import *
```
Please only import what you need, rather than `*`

gundalow commented on lib/ansible/modules/system/beadm.py:
```diff
@@ -0,0 +1,376 @@
+beadm: name=upgrade-be state=activated
+'''
+
+RETURN = '''
```
It looks like this module will return data, if that's correct please document. See other modules for an example

gundalow reviewed (changes requested):
needs_revision 
1) please update CHANGELOG.md in the root if the future repo so the new module is listed under 2.3

2) please add the ANSIBLE_META data before the DOCUMENTION block, see other modules for examples

xen0l committed:
Add ANSIBLE_METADATA
Document return values
Make imports explicit
Use native YAML syntax in EXAMPLES

xen0l committed:
Add beadm module to CHANGELOG.md

xen0l committed:
Update version to 2.3
